request_handlers.cc

```diff
@@ -129,7 +130,21 @@ RequestHandler::Status StaticRequestHandler::handleRequest(const Request& reques
 
   response->SetStatus(Response::OK);
   response->AddHeader("Content-Type", mime_types::extension_to_type(extension));
-  response->SetBody(reply);
+
+  if (extension == "md") {
+    markdown::Document doc;
+    doc.read(reply);
+
+    // markdown::write write out to an ostream, so we need to convert it to a string
+    std::ostringstream stream;
+    doc.write(stream);
+    std::string md_html =  stream.str();
+
+    response->SetBody(md_html);
+  }
+  else {
+    response->SetBody(reply);
```

Comment:
What do you think of switching this if-else order? I personally find it easier to get lost among the Markdown handling code if it happens first. 

Then we'd have: 

```c++
if (!md) {
  // Non-Markdown case
} else {
  // Markdown-handling
}
```